BazelExtensions/extensions.bzl

```diff
@@ -313,7 +330,8 @@ def _make_headermap_impl(ctx):
         if not hasattr(hdr_provider, "objc"):
             continue
 
-        hdrs = hdr_provider.objc.header.to_list()
+        hdrs = hdr_provider.objc.direct_headers
```

Comment:
Can you confirm if this has the same impact as it previously did? I believe here it merges transitive header maps in order to propagate the transitive includes, but would need to look more into that. If it still builds the top 50 swift cocoa pods and passes tests it _should_ work.

Reply:
Good call, looking into it.

Reply:
from the migration description in https://github.com/bazelbuild/bazel/issues/10674, it sounds like the right thing would be to change this to
```
hdrs = hdr_provider[CcInfo].compilation_context.headers.to_list())
```
but that actually fails, while `direct_headers` works fine!

I'm considering 
```
        hdrs = []

        if CcInfo in hdr_provider:
            hdrs.extend(hdr_provider[CcInfo].compilation_context.headers.to_list())

        if hasattr(hdr_provider, "objc"):
            hdrs.extend(hdr_provider.objc.direct_headers)
```

which should cover all bases, but since there's no case of which we're aware that fails with `direct_headers` alone, I'm not sure...

What do you think @jerrymarino?

Reply:
@dgcoffman I'll have to pull this down and take a look at to give a clear answer or on it but - adding direct headers at the end seems right. There are likely some edge cases here which may or may not be exhibited in an integration test. If RN was running on the CI and this PR broke it that's probably something to look out for and vet the logic against. The CI is green ✅ 🚢